src/init_parsing.jl

```diff
@@ -122,14 +145,24 @@ function init_parsing!(io::IO, settings::ParserSettings, options::Parsers.Option
                 if Parsers.sentinel(code)
                     push!(parsing_ctx.header, Symbol(string(settings.default_colname_prefix, i)))
                 elseif !Parsers.ok(code)
-                    close(io);
-                    throw(HeaderParsingError("Error parsing header for column $i at $(lines_skipped_total+1):$(pos) (row:col)."))
+                    close(lexer.io);
+                    throw(HeaderParsingError("Error parsing header for column $i at $(lines_skipped_total+1):$(pos) (row:pos)."))
                 else
                     push!(parsing_ctx.header, Symbol(strip(Parsers.getstring(v, val, options.e))))
                 end
                 pos += tlen
             end
-            !(Parsers.eof(code) || Parsers.newline(code)) && (close(io); throw(HeaderParsingError("Error parsing header, there are more columns than provided types in schema")))
+            if !(Parsers.eof(code) || Parsers.newline(code))
+                ncols = length(parsing_ctx.header)
```

Comment:
this new code is just for the sake of a better error message, right? 
let's add a comment to say why we're doing this extra work
```suggestion
               # There are too many columns; calculate how many extra so we can inform the user.
                ncols = length(parsing_ctx.header)
```